tracsubtickets/web_ui.py

```diff
@@ -246,69 +247,54 @@ def _create_subtickets_table(self, req, children, tbody, depth=0):
-                    link = tag.span('(', link, ')', class_='addsubticket')
+                    # link-specific arguments
+                    script_args['href_req_newticket_with_parent'] \
+                        = req.href.newticket(parents=ticket.id, **inh)
+                    script_args['localized_link_label'] = _('add')
                 else:
-                    inh = [tag.input(type='hidden',
-                                     name=f,
-                                     value=ticket[f]) for f in opt_inherit]
-
-                    button = tag.form(
-                        tag.div(
-                            tag.input(type="submit",
-                                      value=_("Create"),
-                                      title=_("Create a child ticket")),
-                            inh,
-                            tag.input(type="hidden",
-                                      name="parents",
-                                      value=str(ticket.id)),
-                            class_="inlinebuttons"),
-                        method="get", action=req.href.newticket())
-            div.append(button)
-            div.append(tag.h3(_('Subtickets '), link))
-
-        if 'subtickets' in data:
+                    # button-specific arguments
+                    script_args['localized_button_label'] = _("Create")
+                    script_args['localized_button_title'] = \
+                        _("Create new child ticket")
+                    script_args['href_req_newticket'] = req.href.newticket()
+                    script_args['parent_id'] = str(ticket.id)
+                    script_args['inherited_args'] = json.dumps(inh)
+
             # table
-            tbody = tag.tbody()
-            div.append(tag.table(tbody, class_='subtickets'))
-            # tickets
-            self._create_subtickets_table(req, data['subtickets'], tbody)
+            tdict = dict()
+            if 'subtickets' in data:
+                # tickets
+                self._create_subtickets_table(req, data['subtickets'], tdict)
+
+            # subtickets table argument: can be empty but must be set!
+            script_args['subtickets_table']=json.dumps(tdict)
 
-        if div:
             add_stylesheet(req, 'subtickets/css/subtickets.css')
-            '''
-            If rendered in preview mode, DIV we're interested in isn't a child
-            but the root and transformation won't succeed.
-            According to HTML specification, id's must be unique within a
-            document, so it's safe to omit the leading '.' in XPath expression
-            to select all matching regardless of hierarchy their in.
-            '''
-            stream |= Transformer('//div[@id="ticket"]').append(div)
-
-        return stream
+            add_script(req, 'subtickets/js/appendsubticketsdata.js')
+            add_script_data(req, subtickets_script_args=json.dumps(script_args))
```

Comment:
`json.dumps()` and `$.parseJSON()` are not needed. The data is rendered with `to_json` filter in [layout.html](https://github.com/edgewall/trac/blob/trac-1.6/trac/templates/layout.html#L61).